CTX.js

```diff
@@ -70,74 +70,28 @@
  * The approach we will take is to convert this into COinS, so that we can
  * piggy-back off of the perhaps more robust support in the core Zotero code.
  */
+ 
+var items = [];
+var zoteroItems = [];
 
-function detectWeb(doc, url) {
-	var texts = [], text = "";
-	var codes = doc.getElementsByTagName("code");
-	for(var i = 0; i < codes.length; i++) {
-		text = codes[i].textContent;
-		text.replace(/</g,"&lt;").replace(/>/g,"&gt;");
-		texts.push(text);
-	}
-	return detectInString(texts);
-};
-
-function doWeb(doc, url) {
-	var texts = [], text = "";
-	var codes = doc.getElementsByTagName("code");
-	for(var i = 0; i < codes.length; i++) {
-		text = codes[i].textContent;
-		text.replace(/</g,"&lt;").replace(/>/g,"&gt;");
-		texts.push(text);
-	}
-	doImportFromText(texts, true);
-};
-
-function doImport() {
+function detectImport() {
 	var text = "";
 	var line;
 	while(line = Zotero.read()) {
 		text += line;
```

Comment:
limit this to something like 100 lines and you can return as soon as that string appears. I'm also fairly certain that the string below can never be split over multiple lines, so you don't even have to concatenate.